source/rafcon/design_patterns/singleton.py

```diff
@@ -16,6 +16,14 @@ def instance(self, *args, **kwargs):
             self._instance = self._cls(*args, **kwargs)
         return self._instance
 
+    def reset_and_instance(self, *args, **kwargs):
```

Comment:
Do you have another solution?

Reply:
I think that singletons are not a very good pattern, as you can see in this case.

But the singletons are new there, so I propose to add a `reset()` method to the GVM that removes all variables. Instead of calling then `reset_and_instance()`, you first call `instance()` and then `reset()` on the instance.

The question is if not other singletons also hold some state which should be reset. The `ExecutionEngine` probably holds a state machine, etc.